service/debugger/debugger.go

```diff
@@ -1464,6 +1467,82 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
+
+	idx := strings.Index(f.Name, "runtime.")
+	idx2 := strings.Index(f.Name, "runtime.defer")
+	if idx != -1 && idx2 == -1 {
+		// Except defer functions do not traverse runtime.* functions
+		return nil, nil
+	}
+
+	if depth > FollowCalls {
+		// If we already reached desired depth, return
+		return nil, nil
+	}
+	funcs := []string{}
+	funcs = append(funcs, f.Name)
+	text, err := proc.Disassemble(t.Memory(), nil, t.Breakpoints(), t.BinInfo(), f.Entry, f.End)
+	if err != nil {
+		return nil, fmt.Errorf("disassemble failed with error %s", err.Error())
+	}
+	depth++
+	for _, instr := range text {
+		if instr.IsCall() && instr.DestLoc != nil && instr.DestLoc.Fn != nil {
+			cf := instr.DestLoc.Fn
+			if cf.Name == f.Name {
+				continue
+			}
+			if depth <= FollowCalls {
+				children, err := traverse(t, cf, depth, FollowCalls)
+				funcs = append(funcs, children...)
+				if err != nil {
+					return nil, fmt.Errorf("disassemble failed with error %s", err.Error())
+				}
+			}
+		}
+	}
+	for _, fbinary := range t.BinInfo().Functions {
```

Comment:
Can we optimize this by building an index as we go instead of looping through all program functions again? This doesn't necessarily have to happen in this PR, but perhaps it would be better to represent all program functions using a tree data structure as opposed to a slice. Let's do this later actually, but something to keep in mind.